modules/ballerina-core/src/main/java/org/ballerinalang/model/util/XMLUtils.java

```diff
@@ -159,6 +159,7 @@
                 concatSeq.add(j++, seq.get(i));
```

Comment:
I think each `seq.get(i)` should also get built. We can implement the `build()` in XMLSequence as well, which will recursively visit its children and build them.

Then we can call `firstSeq.build();` and `secondSeq.build();` at the start of this (`concatentae(...)`) method.. WDYT?

Reply:
Fixed. Thanks.